kernel/nano_core/build.rs

```diff
@@ -33,20 +28,24 @@ cfg_if::cfg_if! {
 #[cfg(all(feature = "uefi", feature = "bios"))]
 compile_error!("Both the 'bios' and 'uefi' features cannot be jointly enabled");
 
-/// The set of built-in environment variables defined by cargo.
+/// The prefix that all custom rustc-known cfg keys are given by cargo
+/// when it transforms them into environment variables.
+const CARGO_CFG_PREFIX: &str = "CARGO_CFG_";
+
+/// The set of built-in environment variables defined by cargo, without prefix.
 static NON_CUSTOM_CFGS: [&str; 12] = [
-    "CARGO_CFG_PANIC",
-    "CARGO_CFG_TARGET_ABI",
-    "CARGO_CFG_TARGET_ARCH",
-    "CARGO_CFG_TARGET_ENDIAN",
-    "CARGO_CFG_TARGET_ENV",
-    "CARGO_CFG_TARGET_FEATURE",
-    "CARGO_CFG_TARGET_HAS_ATOMIC",
-    "CARGO_CFG_TARGET_HAS_ATOMIC_EQUAL_ALIGNMENT",
-    "CARGO_CFG_TARGET_HAS_ATOMIC_LOAD_STORE",
-    "CARGO_CFG_TARGET_OS",
-    "CARGO_CFG_TARGET_POINTER_WIDTH",
-    "CARGO_CFG_TARGET_VENDOR",
+    "PANIC",
+    "TARGET_ABI",
+    "TARGET_ARCH",
+    "TARGET_ENDIAN",
+    "TARGET_ENV",
+    "TARGET_FEATURE",
+    "TARGET_HAS_ATOMIC",
+    "TARGET_HAS_ATOMIC_EQUAL_ALIGNMENT",
+    "TARGET_HAS_ATOMIC_LOAD_STORE",
+    "TARGET_OS",
+    "TARGET_POINTER_WIDTH",
+    "TARGET_VENDOR",
```

Comment:
i find this to be less clear / worse, since the reader cannot quickly observe and understand that we're searching for default/built-in `CARGO_CFG_*` values. Please revert.

Reply:
It's worse for grepping I guess. I did that for dedup and code flow reasons below. Will revert.